zero/sources_explicit.c

```diff
@@ -105,32 +106,73 @@ explicit_frictional_source_update(const gkyl_moment_em_coupling* mom_em, double
     double T_elc = mom_em->friction_T_elc;
     double Lambda_ee = mom_em->friction_Lambda_ee;
 
-    double f_elc_new[5], f_elc_stage1[5], f_elc_stage2[5];
-    double f_ion_new[5], f_ion_stage1[5], f_ion_stage2[5];
-    double f_elc_old[5], f_ion_old[5];
+    int elc_num_equations = 0;
+    int ion_num_equations = 0;
 
-    for (int i = 0; i < 5; i++) {
+    if (mom_em->param[0].type == GKYL_EQN_EULER) {
+      elc_num_equations = 5;
+    }
+    else if (mom_em->param[0].type == GKYL_EQN_ISO_EULER) {
+      elc_num_equations = 4;
+    }
+
+    double *f_elc_new = gkyl_malloc(sizeof(double[elc_num_equations]));
```

Comment:
We need to avoid these mallocs here. Otherwise our performance can be impacted. We should just make these arrays on the stack since they're small.